File: Simple-Encryption-1-Alternating-Split/0.java
public class Kata {
    public static String encrypt(final String text, final int n) {
        if (n <= 0) return text;
        var buffer = new char[text.length()];
        var offset = text.length() / 2;
        for (int i = 0; i < text.length(); ++i) {
            if (i % 2 == 1) buffer[i / 2] = text.charAt(i);
            else buffer[i / 2 + offset] = text.charAt(i);
        }
        return encrypt(new String(buffer), n - 1);
    }

    public static String decrypt(final String text, final int n) {
        if (n <= 0) return text;
        var buffer = new char[text.length()];
        var offset = text.length() / 2;
        for (int i = 0; i < text.length(); ++i) {
            if (i % 2 == 1) buffer[i] = text.charAt(i / 2);
            else buffer[i] = text.charAt(i / 2 + offset);
        }
        return decrypt(new String(buffer), n - 1);
    }
}

File: 57814d79a56c88e3e0000786/0.hs
module AlternateSplit.JorgeVS.Kata where

import Data.List

encrypt :: String -> Int -> String
encrypt s n
  | n <= 0    = s
  | otherwise = encrypt s' (n - 1)
  where
    s' = map fst . uncurry (++) $ partition (odd . snd) (zip s [0..])

decrypt :: String -> Int -> String
decrypt s n
  | n <= 0    = s
  | otherwise = decrypt s' (n - 1)
  where
    s' = concat . transpose . (\(a, b) -> [b, a]) $ splitAt (length s `div` 2) s
